### PR TITLE
Patch for Irregular Incoming INIT Ballot

### LINKED ISSUE #520: Recent Integration-Test Failure
Symtoms
* Integration test is failed too often, thereagainst successes are rare.
* Integration test is failed in CircleCI and also local env.
* The failed test case is randomly changed.
* Too many EXPIRED ballot.

More details story and analysis of logs can be found at https://gist.github.com/spikeekips/391287344a91e1a9f8950c5ca32570eb .

### PR DESCRIPTION
**Pull request by spikeekips**
### Github Issue

resolves #520 

### Background

Check #520 .

### Solution

The incoming well-formed `INIT` ballot should be broadcasted.

### Integration Test Result

In my local env, I tests this patch many times and result was good.
```
================================================================================
+ true
+ docker rm -f 4ba28c692f9bdcf802f8f185d6f7ddaab23d0c7d82dc804a41696c665fc999bb bd97584983b1875073b24de0b3f677227126ac7b043a92fc5d8f2773f7390950 ff26e05ef46737b9585e6205298dc20ceb63a303cc1e40c463ca5b1bce723a8f 6d64314b3b3f9efb262ea16abce1ce406b869728e3d18ea2b3f82d2cd4a6e9bc 042c7720cf2b8aec257075541b1fef1e37d6684e878a0afdea90883a5ef56881 afb17990572c04702e7f64fd92bdb3e134e947d4719d60bda12b29003966530f
Error: No such container: 4ba28c692f9bdcf802f8f185d6f7ddaab23d0c7d82dc804a41696c665fc999bb
================================================================================
+ true
+ docker rm -f 7c277a9035e51aa74e4e440772cb396e06f0902db6f9a0c86356fa053e564f3c 2d67511460d10571c196ad0230492c8f0e0c3b975586bc2479fef5bbe805aa5c fc43e68661e4e70476d63668c77dc9407e8748b051f28d23114a3fb796809a9b b697955d7b17972994fdf3862b15b7ca60d29456c935423230d7631c5837bc3a f5c4f73ad62a276594c1878248d88c127674a073c47643b45048eed8ca527c6f 6b7202daeed05c8478421a63a6886055e64252c331ef7abdfca4b2a1ac9dea78
Error: No such container: 7c277a9035e51aa74e4e440772cb396e06f0902db6f9a0c86356fa053e564f3c
================================================================================
+ true
+ docker rm -f e71bbdbb88348c99bd56c4cc7ab43680d02a753d5294b0b63466eeb2079c91ec 8a273b44fbef18f6a29de62089b6284976b783bec24d0c8290279bf4eda6fd0d 52c84767b2563fe5295c7c378523088c22899b0c65ed51a70f59ccce74a12956 1aabfaae96f6f6933b322bc88f3a20a24973cbae87ea67030a7dd438078bd417 8f191a3a6943a448e8d69c0b912f5e979d9599a6cdc3b032cbee6be5de85c70f 65a073fee2be6366e7736d3503cbe00369699b11de060af96eaadbdb088bc16b
Error: No such container: e71bbdbb88348c99bd56c4cc7ab43680d02a753d5294b0b63466eeb2079c91ec
================================================================================
+ true
+ docker rm -f e407a34811e1c441a6437494bb9ab8e154e46e71ea76a6f072944dbfee0308da ba9534fa1b8064ccca89e52c2ce520248fdf955b51d9b4ffe4ea5f75c3d7cf71 5a3dafd7cf8255adbbb3ec25baac8c6de0c5a83fee8c27166d8d7af2724374c5 bb2a25205e424d8371553e5f897ea98c26d9302dd32102a3573a59bd5e3fc593 0fc8943c673d08c65b2f9150ef8023064aba7c98ea46dc1514bbcca32a7388db 4c671cabe3e0bd08750f8d675b4dcce9d5812c11af0c6d9fd9b91ab74df63c9b
Error: No such container: e407a34811e1c441a6437494bb9ab8e154e46e71ea76a6f072944dbfee0308da
================================================================================
+ true
+ docker rm -f 16f79070309cebdf056a3fb905719d5b80519dd34dc9eae1f0398edf436caa03 013446acd3ca71af24329d64e9439dbaf27750dc9fa3745c156c98b20eaf5a80 2048f4366ecbace9e32dd4d6568576091b9db017e323459917d55393a89ab4e5 27c5faeb88d7fda0c461b729a9da0ceaf489cf1cca64f97a83a55fa607f8aedf e5d812e52d39be5d727d58a96eb5e30d49076bf23e0583f2db10b4991dee7d3a 93f01ab165d83473fd05e2b52e6dbc1c98a28fc48376f3a2f5f85176a18ad0d9
Error: No such container: 16f79070309cebdf056a3fb905719d5b80519dd34dc9eae1f0398edf436caa03
================================================================================
+ true
+ docker rm -f 6cedebbc3abd1ed49f124a90b4b47d9331f13b33abfc95f03e2de697ff6e84f9 e0303f4adbd7b1890cdc12584049c32a6de8e71a49219e9d9db26874becb9ced 7ca84d811ac5ce796166093f8367691c7abb644c389906cf61070ba787e6ea39 e8d1bc3c3cfce8c63d0ab54dc4fd937d236dcbb8501c14b5e268db47f90c8347 b28f8fb50210b1ac7eae12aa458a2ce15619ba2470bc802da92798ac9638b1c2 5e3f824514246999a4982ce09e8499c78a40d54db44c7fa2ad7fa5d7594c6a83
Error: No such container: 6cedebbc3abd1ed49f124a90b4b47d9331f13b33abfc95f03e2de697ff6e84f9
================================================================================
+ true
+ docker rm -f 68c0ef0de7d6674609b267d5133024b5543fed824219235b84df609cb5040bba 9f2a2dbf25a4d84aac9abcc09e9c42e255a8f8fe9d119e478df06c47f80b2f17 0e2d688175775cb23fb86e818043391ffc6885229cf01f29bef08c8db4ff57d9 bd0dd1150ebc9af5f9147a4357ab6ce42a5ec60e31ffcb919cb06a6ccdf062f4 6d16211ec53e8b105de6466ca44fbdd2d31ccbc6cc57293938d9315b384dc5f2 ddce11549a4119feed06e6f6d21c44d384f9ce800ee73849ed48c2bff49a907d
Error: No such container: 68c0ef0de7d6674609b267d5133024b5543fed824219235b84df609cb5040bba
================================================================================
+ true
+ docker rm -f 1779c785ba71441059308a9665462f6678c21547b606fd1d0db98f7583651cff 2d9ffc4a3697add53709f05f8924fc08d6c18cceadfc4619daa734119e3ee8ae 2a93bbc36a42a0f20c031f5e9d520fabc23476c2f2743e47aa08f2bbbaf0f77e 8533740b76ff59efcd1a48392374a809cb0e3fd43d13550f8a9eed7e82cf3342 ddb0764b2c303d5c0ff85220fc9d19e542d5694986dd831d4b5c91b08155d250 5aa68bb088f6d6af90136ecb2e4f9e4e081b4412b03b9d5d5ac3a9fde9280078
Error: No such container: 1779c785ba71441059308a9665462f6678c21547b606fd1d0db98f7583651cff
================================================================================
+ true
+ docker rm -f 8aabdd08a2c0924f5170153682dffde2da133690fdac2329a8dc7488ff2c17d4 89a2125fa5d5b39fdbc986bca0dbabf237ca74692375bece3e769c75214f080c 233c59047ecc2ef24e1b06929996cb2c1707d0509b61d3b27024da44da601694 dc36a2a1eb3945c72040634119415ec82147fd07a481774cc4609cb716ff3aaa 71bda1fec34c7709c9e4ecc1ea26bb276b7271023719c6c894749ca6127a8ea3 b5de4df82c67845a72b918074edb40b22a9ee659eff3e033a6adc17d668d842e
Error: No such container: 8aabdd08a2c0924f5170153682dffde2da133690fdac2329a8dc7488ff2c17d4
================================================================================
+ true
+ docker rm -f 9742954405fecc5577208bcc7d744837f9995d8f4306d0568fcac692b2457270 4dfba575f727c4d8b81d0ddbcd00668781f4b876b19527c059869ff386997b90 4d312d18c17fe41bff82a37b1a9f29882e2dfb158df065fe853bea6aeec595cf 8f7d8a98a9ea816dae6b6bf3d9d8f9a6c1e0194b13827aee82ae3853f7514a45 7291e4f61a9834a893b05a10d8a817d0dd2e18bf1bec48248fe0460a9fef0d52 64f5e9ba8e3898c5ebe1f9430fd8b7be44b3cc7b1ec41312dc6b7a08878ba8cd
Error: No such container: 9742954405fecc5577208bcc7d744837f9995d8f4306d0568fcac692b2457270
```

### CODE REVIEW

File: lib/node/runner/checker.go [hunk 1]
```diff
@@ -6,8 +6,6 @@ import (
 	"encoding/json"
 	"io"
 
-	logging "github.com/inconshreveable/log15"
```
Comment (Charleslee522) [inline lib/node/runner/checker.go]: @spikeekips I think this position is better because of code consistency.

Comment (spikeekips) [reply lib/node/runner/checker.go]: I also wanted, but I updated my ‘vim-go’ and then new `goimports` forced to rearrange like that :(

Comment (Charleslee522) [reply lib/node/runner/checker.go]: Oh, then nevermind :)

**Comment by Charleslee522** [comment]
I want to give you 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍 👍

File: lib/node/runner/checker_test.go [hunk 2]
```diff
@@ -343,3 +343,162 @@ func TestGetMissingTransactionProposerAlsoMissing(t *testing.T) {
 	require.Equal(t, ballot.VotingNO, checker.VotingHole)
 	require.Equal(t, 0, g.consensusNR.Consensus().TransactionPool.Len())
 }
+
+type IrregularIncomingBallot struct {
```
Comment (Geod24) [inline lib/node/runner/checker_test.go]: Why is it public ?

Comment (spikeekips) [reply lib/node/runner/checker_test.go]: @Geod24 No reason, I don't know :)

Comment (Geod24) [reply lib/node/runner/checker_test.go]: Then make it private

File: lib/node/runner/checker_test.go [hunk 3]
```diff
@@ -343,3 +343,162 @@ func TestGetMissingTransactionProposerAlsoMissing(t *testing.T) {
...
+
+	genesisBlock   block.Block
+	commonAccount  *block.BlockAccount
+	initialBalance common.Amount
+
+	keyA     *keypair.Full
+	accountA *block.BlockAccount
+}
+
+func (p *IrregularIncomingBallot) Prepare() {
```
Comment (Geod24) [inline lib/node/runner/checker_test.go]: Should probably also be private if you don't know why it's public

**Pull request merged**